homeassistant/components/nuki/config_flow.py

```diff
@@ -92,33 +89,51 @@ async def async_step_dhcp(self, discovery_info: dict):
     async def async_step_reauth(self, data):
         """Perform reauth upon an API authentication error."""
         self._data = data
-        self._reauth = True
 
         return await self.async_step_reauth_confirm()
 
     async def async_step_reauth_confirm(self, user_input=None):
         """Dialog that inform the user that reauth is required."""
+        errors = {}
         if user_input is None:
             return self.async_show_form(
                 step_id="reauth_confirm", data_schema=REAUTH_SCHEMA
             )
-        return await self.async_step_validate(user_input)
+
+        conf = {
+            CONF_HOST: self._data[CONF_HOST],
+            CONF_PORT: self._data[CONF_PORT],
+            CONF_TOKEN: user_input[CONF_TOKEN],
+        }
+
+        try:
+            info = await validate_input(self.hass, conf)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+
+        if "base" not in errors:
+            existing_entry = await self.async_set_unique_id(info["ids"]["hardwareId"])
+            if existing_entry:
```

Comment:
Should we handle the other case and show an error that no matching entry was found?

Reply:
In the [docs](https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#reauthentication) it's not handling the other case, if I should implement this, should it abort the flow or just print an error?

Reply:
The example in the docs creates a new entry if there's no existing entry. Here we just show the form again without any error message. How will the user know what to do?

Reply:
Oh yeah, I've misread that in the docs. I've added the "unknown" error since that is the only one from the common strings that would match here.

Reply:
We can add a custom message if we have better info than unknown.